cognite/client/data_classes/contextualization.py

```diff
@@ -1036,3 +1038,176 @@ def __init__(
 
 class ResourceReferenceList(CogniteResourceList[ResourceReference]):
     _RESOURCE = ResourceReference
+
+
+class NestableDiagramDetectConfig:
```

Comment:
Let's make these classes extend `CogniteObject`.

Reply:
I answered in more detail in a different thread, but it doesn't seem like a good fit.

Although I just remembered the API returns this config as part of the detect job result 🤔 
Maybe we could load the results into this object then. I'll investigate further.